test/test_train_cifar.py

```diff
@@ -154,11 +154,12 @@ def train_cifar():
 
   def train_loop_fn(model, loader, device, context):
     loss_fn = nn.CrossEntropyLoss()
-    optimizer = optim.SGD(
-        model.parameters(),
-        lr=FLAGS.lr,
-        momentum=FLAGS.momentum,
-        weight_decay=5e-4)
+    optimizer = context.getattr_or(
+        'optimizer', lambda: optim.SGD(
```

Comment:
Might want to get rid of the lambda? What do you think?

Reply:
A lambda allows to defer creation, and only create once.
Otherwise every call to it will create a new instance, which will then be discarded.

Reply:
Yeah, of course, but I thought not using lambdas would make it more readable and discarded instances wouldn't be a big deal since they wouldn't have any references and be garbage collected anyways. But yes, leaving it up to you.

Reply:
I would personally favor adding "lambda:" characters over creating new N optimizers and throw away N-1 of them.
It is not a matter of performance, but it looks saner to me.
And if optimizers creations have side effects, that becomes an issue as well.
Keep in mind that folks look at these examples and use them as API examples, so IMO better some extra "lambda:" (which they can happily remove if they see it fit).